aviary/subsystems/propulsion/turboprop_model.py

```diff
@@ -78,116 +77,205 @@ def __init__(self, name='turboprop_model', options: AviaryValues = None,
     # BUG if using both custom subsystems that happen to share a kwarg but need different values, this breaks
     def build_pre_mission(self, aviary_inputs, **kwargs) -> om.Group:
         shp_model = self.shaft_power_model
-        prop_model = self.propeller_model
+        propeller_model = self.propeller_model
         turboprop_group = om.Group()
         # TODO engine scaling for turboshafts requires EngineSizing to be refactored to
         # accept target scaling variable as an option, skipping for now
         if type(shp_model) is not EngineDeck:
             shp_model_pre_mission = shp_model.build_pre_mission(aviary_inputs, **kwargs)
             if shp_model_pre_mission is not None:
-                turboprop_group.add_subsystem(shp_model_pre_mission.name,
-                                              subsys=shp_model_pre_mission,
-                                              promotes=['*'])
-
-        if prop_model is not None:
-            prop_model_pre_mission = prop_model.build_pre_mission(
-                aviary_inputs, **kwargs)
-            if prop_model_pre_mission is not None:
-                turboprop_group.add_subsystem(prop_model_pre_mission.name,
-                                              subsys=prop_model_pre_mission,
-                                              promotes=['*'])
+                turboprop_group.add_subsystem(
+                    shp_model_pre_mission.name,
+                    subsys=shp_model_pre_mission,
+                    promotes=['*']
+                )
+
+        if propeller_model is not None:
+            propeller_model_pre_mission = propeller_model.build_pre_mission(
+                aviary_inputs, **kwargs
+            )
+            if propeller_model_pre_mission is not None:
+                turboprop_group.add_subsystem(
+                    propeller_model_pre_mission.name,
+                    subsys=propeller_model_pre_mission,
+                    promotes=['*']
+                )
 
         return turboprop_group
 
     def build_mission(self, num_nodes, aviary_inputs, **kwargs):
         shp_model = self.shaft_power_model
-        prop_model = self.propeller_model
+        propeller_model = self.propeller_model
         turboprop_group = om.Group()
+        max_thrust_group = om.Group()
 
-        shp_model_mission = shp_model.build_mission(num_nodes, aviary_inputs, **kwargs)
+        try:
+            shp_kwargs = kwargs[shp_model.name]
```

Comment:
I might be thinking about this wrong, but if you grab the `kwargs[shp_model.name]`, is that a dict object? Or something else? I couldn't quite figure out how or where this logic is used

Reply:
It's how you pass kwargs down to models inside turboprop model- this is checking if you have any kwargs for models that match the name of the shaft power model, and taking that and passing it to `build_mission()`